ios/RNTypeValidation.m

```diff
@@ -0,0 +1,211 @@
+    }
+    
+    return nil;
+}
+
++ (nullable NSString *)isApplePayConfigurationValid: (NSMutableDictionary *) properties {
+    NSDictionary *configuration = properties[@"configuration"];
+    NSString *isConfigurationValid = [self isConfigurationValid:properties];
+    if (isConfigurationValid) {
+        return isConfigurationValid;
+    }
+    if (![configuration[@"applePayConfiguration"] isKindOfClass: [NSMutableDictionary class]]) {
+        return @"applePayConfiguration is not valid, make sure applePayConfiguration corespond to JPApplePayConfiguration model";
+    } else {
+        NSMutableDictionary *applePayConfiguration = configuration[@"applePayConfiguration"];
+        
+        if (![applePayConfiguration[@"merchantId"] isKindOfClass:[NSString class]]) {
+            return (@"applePayConfiguration -> merchantId is not valid, it should be a string");
+        }
+        
+        if(![applePayConfiguration[@"countryCode"] isKindOfClass:[NSString class]]) {
+            return(@"applePayConfiguration -> countryCode is not valid, it should be a string");
+        }
+        
+        if(![self isPaymentSummaryItemsValid:applePayConfiguration[@"paymentSummaryItems"]]) {
+            return (@"applePayConfiguration -> paymentSummaryItems is not valid, it should be an array of PaymentSummaryItems type");
+        }
+        
+        if (![applePayConfiguration[@"merchantCapabilities"] isKindOfClass:[NSNumber class]]) {
+            return @"applePayConfiguration -> merchantCapabilities is not valid, it should be a number";
+        }
+        
+        if(![applePayConfiguration[@"requiredBillingContactFields"] isKindOfClass:[NSNumber class]]) {
+            return  @"applePayConfiguration -> requiredBillingContactFields is not valid, it should a number";
+        }
+        
+        if(![applePayConfiguration[@"requiredShippingContactFields"] isKindOfClass:[NSNumber class]]) {
+            return @"applePayConfiguration -> requiredShippingContactFields is not valid, it should be a number";
+        }
+        
+        if(![self isShippingMethodsValid:applePayConfiguration[@"shippingMethods"]]) {
+            return @"applePayConfiguration -> shippingMethods is not valid, it should be an array of ShippingMethods type";
+        }
+        
+        if(![applePayConfiguration[@"shippingType"] isKindOfClass:[NSNumber class]]) {
+            return @"applePayConfiguration -> shippingType is not valid it  should be a number";
+        }
+        
+        if(![applePayConfiguration[@"returnedInfo"] isKindOfClass:[NSNumber class]]){
+            return @"applePayConfiguration -> returnedInfo is not valid, it should be a number";
+        }
+        
+        return nil;
+    }
+}
+
+//---------------------------------------------------
+// MARK: - Helper methods
+//---------------------------------------------------
+
++ (BOOL)isAmountDictionaryValid:(NSMutableDictionary *)amount {
```

Comment:
Most of these methods should probably return an optional NSString describing the error cause